strings/duration_gen.go

```diff
@@ -1,105 +1,42 @@
-// Code generated by gencel. DO NOT EDIT.
-
 package strings
 
-import "github.com/google/cel-go/cel"
-import "github.com/google/cel-go/common/types"
-import "github.com/google/cel-go/common/types/ref"
-
-var durationStringGen = cel.Function("duration.String",
-	cel.Overload("duration.String_",
-		nil,
-		cel.StringType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.String())
-
-		}),
-	),
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
 )
 
-var durationNanosecondsGen = cel.Function("duration.Nanoseconds",
-	cel.Overload("duration.Nanoseconds_",
-		nil,
-		cel.IntType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.Nanoseconds())
-
+// var durationHumanize = cel.Function("HumanDuration",
+// 	cel.Overload("duration.HumanDuration",
+// 		[]*cel.Type{cel.DynType},
+// 		cel.StringType,
+// 		cel.UnaryBinding(func(value ref.Val) ref.Val {
+// 			return types.String(HumanDuration(value.Value()))
+// 		}),
+// 	),
+// )
+
+var durationAgeGen = cel.Function("Age",
+	cel.Overload("duration.Age",
+		[]*cel.Type{cel.StringType},
+		cel.DurationType,
+		cel.UnaryBinding(func(value ref.Val) ref.Val {
+			a := Age(value.Value().(string))
+			return types.Duration{Duration: a}
 		}),
 	),
 )
 
-var durationSecondsGen = cel.Function("duration.Seconds",
-	cel.Overload("duration.Seconds_",
-		nil,
-		cel.DoubleType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.Seconds())
-
-		}),
-	),
-)
-
-var durationHoursGen = cel.Function("duration.Hours",
-	cel.Overload("duration.Hours_",
-		nil,
-		cel.DoubleType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.Hours())
-
-		}),
-	),
-)
-
-var durationDaysGen = cel.Function("duration.Days",
-	cel.Overload("duration.Days_",
-		nil,
-		cel.DoubleType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.Days())
-
-		}),
-	),
-)
-
-var durationWeeksGen = cel.Function("duration.Weeks",
-	cel.Overload("duration.Weeks_",
-		nil,
-		cel.DoubleType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.Weeks())
-
-		}),
-	),
-)
-
-var durationMinutesGen = cel.Function("duration.Minutes",
-	cel.Overload("duration.Minutes_",
-		nil,
-		cel.DoubleType,
-		cel.FunctionBinding(func(args ...ref.Val) ref.Val {
-
-			var x Duration
-
-			return types.DefaultTypeAdapter.NativeToValue(x.Minutes())
-
+var durationDurationGen = cel.Function("Duration",
```

Comment:
@moshloop The native cel `duration` func doesn't support days. so I had to add our own `Duration` func back.